app/browser/tabs.js

```diff
@@ -283,6 +284,57 @@ const updateAboutDetails = (tab, tabValue) => {
   }
 }
 
+// hack to deal with about:* pages
+const fixDisplayURL = (navigationEntry, controller) => {
+  if (navigationEntry == null) {
+    return null
+  }
+
+  navigationEntry = Object.assign({}, navigationEntry)
+
+  navigationEntry.virtualURL = getLocationIfPDF(navigationEntry.virtualURL)
+
+  if (isTargetAboutUrl(navigationEntry.virtualURL)) {
+    navigationEntry.virtualURL = getSourceAboutUrl(navigationEntry.virtualURL)
+  }
+
+  if (isIntermediateAboutPage(navigationEntry.virtualURL)) {
+    const previousEntry = controller.getEntryAtOffset(-1)
+    if (!controller.canGoForward() && previousEntry) {
+      navigationEntry.virtualURL = previousEntry.virtualURL
+    }
+  }
+
+  if (isTargetMagnetUrl(navigationEntry.virtualURL)) {
+    navigationEntry.virtualURL = getSourceMagnetUrl(navigationEntry.virtualURL)
+  }
+
+  if (navigationEntry.virtualURL === 'about:newtab') {
+    navigationEntry.virtualURL = ''
+  }
+
+  navigationEntry.virtualURL = muon.url.formatForDisplay(navigationEntry.virtualURL)
+
+  const parsedURL = muon.url.parse(navigationEntry.virtualURL)
+  navigationEntry = Object.assign(parsedURL, navigationEntry)
+
+  return navigationEntry
+}
+
+const createNavigationState = (navigationHandle, controller) => {
+  return Immutable.Map({
+    hasCommitted: navigationHandle.hasCommitted(),
+    isErrorPage: navigationHandle.isErrorPage(),
+    netErrorCode: navigationHandle.getNetErrorCode(),
+    activeEntry: fixDisplayURL(controller.getActiveEntry(), controller),
```

Comment:
This would be a good place for some comments about what each *Entry property is for.  Please add in a follow up.